core/common/src/IModelError.ts

```diff
@@ -32,6 +32,43 @@ export class IModelError extends BentleyError {
   }
 }
 
+/** Detailed information about a particular object Lock that is causing the Lock update conflict.
+ * @public
+*/
+export interface ConflictingLock {
+  /** Id of the object that is causing conflict. */
+  objectId: string;
+  /**
+     * The level of conflicting lock. Possible values are {@link LockLevel.Shared}, {@link LockLevel.Exclusive}.
+     * See {@link LockLevel}.
+     */
+  lockLevel: LockLevel;
+  /** An array of Briefcase ids that hold this lock. */
+  briefcaseIds: number[];
+}
+
+/** Supported Lock levels.
+ * @public
+*/
+export enum LockLevel {
+  /** Objects are not locked. This level is used for releasing already acquired locks. */
+  None = "none",
+  /** Multiple Briefcases can acquire a shared lock on the same object. */
+  Shared = "shared",
+  /** Only one Briefcase can acquire an exclusive lock on a given object at a time. */
+  Exclusive = "exclusive"
+}
+
+/** @public */
```

Comment:
Public APIs must be documented.

Reply:
I added a comment to ConflictingLOcksError